src/cookie.ts

```diff
@@ -0,0 +1,67 @@
+import type { IncomingMessage, ServerResponse } from 'http'
+
+export interface CookieOptions {
+  expires: Date
+  maxAge: number
+  domain: string
+  path: string
+  secure: boolean
+  httpOnly: boolean
+  sameSite: 'Strict' | 'Lax' | 'None'
+}
+
+export function isValidCookieName (key: string): boolean {
+  return !key.match(/\(\)<>@,;:"\/\[\]\?=\{\}/)
+}
+
+export function useCookie (req: IncomingMessage): Promise<string> {
+  // @ts-ignore
+  if (req.cookie) {
+    // @ts-ignore
+    return req.cookie
+  }
+
+  const { cookie } = req.headers
+  const cookieEnteries = (cookie || '')
+    .split(';')
+    .map((item) => {
+      const [key, value] = item.trim().split('=')
+      return [key, decodeURI(value)]
+    })
+
+  const cookieObject = Object.fromEntries(cookieEnteries)
+
+  // @ts-ignore
+  req.cookie = cookieObject
```

Comment:
I think we can add an TS assertion that extends the Request object once this function has been called...

Reply:
Cache value shouldn't be exposed or implicitly directly accessed.